src/Rules/Variables/UnsetRule.php

```diff
@@ -69,6 +76,35 @@ private function canBeUnset(Node $node, Scope $scope): ?IdentifierRuleError
 			}
 
 			return $this->canBeUnset($node->var, $scope);
+		} elseif (
+			$node instanceof Node\Expr\PropertyFetch
+			&& $node->name instanceof Node\Identifier
+		) {
+			$foundPropertyReflection = $this->propertyReflectionFinder->findPropertyReflectionFromNode($node, $scope);
+			if ($foundPropertyReflection === null) {
+				return null;
+			}
+
+			$propertyReflection = $foundPropertyReflection->getNativeReflection();
+			if ($propertyReflection === null) {
+				return null;
+			}
+
+			if ($propertyReflection->isReadOnly() || $propertyReflection->isReadOnlyByPhpDoc()) {
+				$type = $scope->getType($node->var);
+
+				return RuleErrorBuilder::message(
+					sprintf(
+						'Cannot unset %s property %s of %s.',
```

Comment:
Usually we do Foo::$name format.

%s of %s sounds like an ancient text: Yennefer of Vengerberg 😀